source/Ship.cpp

```diff
@@ -64,31 +64,32 @@ namespace {
 
 	const double MAXIMUM_TEMPERATURE = 100.;
 
-	// Scanning takes between 2 and 10 seconds (SCAN_TIME / MAX_SCAN_STEPS and SCAN_TIME / MIN_SCAN_STEPS)
+	// Scanning takes up to 10 seconds (SCAN_TIME / MIN_SCAN_STEPS)
 	// dependent on the range from the ship (among other factors).
 	// The scan speed uses a gaussian drop-off with the reported scan radius as the standard deviation.
 	const double SCAN_TIME = 600.;
 	// Always gain at least 1 step of scan progress for every frame spent scanning while in range.
 	// This ensures every scan completes within 600 frames (10 seconds) of being in range.
 	const double MIN_SCAN_STEPS = 1;
-	// Gain no more than 5 steps of scan progress for every frame spent scanning while in range.
-	// This ensures no scan takes fewer than 120 frames (2 seconds.)
-	const double MAX_SCAN_STEPS = 5; // minimum of 2 seconds to scan
+	// Exponential dropoff of scan rate starts at scan rate of SCAN_TIME/LINEAR_RATE
+	// with an exponent of SCAN_DROPOFF_EXPONENT.
+	const double LINEAR_RATE = 5;
+	const double SCAN_DROPOFF_EXPONENT = 5;
 
 	// These numbers ensure it takes 10 seconds for a Cargo Scanner to scan
 	// a Bulk Freighter at point blank range. Any ship with less than 40
 	// cargo space takes as long as a ship with 40 cargo space.
 	const double SCAN_MIN_CARGO_SPACE = 40;
-	const double SCAN_CARGO_FACTOR = 3;
+	const double SCAN_CARGO_FACTOR = pow(1.0 / sqrt(5), -1.5) / 600;
```

Comment:
This can be simplified. Although, it might be a good idea to calculate the final value directly and use that instead.

Reply:
No, no, definitely not. That's how I messed it up before. With the numbers present, you see: 5=cargo scanner, 600=bulk freighter size. Then you compare it to the formula a few lines down and you know you got it right.

Reply:
IMO if you add a comment saying which formula was used I think that would be enough. There's really no need to calculate this value at runtime.

Reply:
I've rearranged the comments so it makes more sense. How does it look now?

Reply:
> There's really no need to calculate this value at runtime.

It won't be. The compiler will evaluate this at compile time. 

https://godbolt.org/z/qbva75sbe

Reply:
I think it's one of those things where the compiler isn't *required* to optimize it away, so you're a slave to the optimization heuristics. Maybe there's another way to express it that would force MSVC to optimize it away?

Reply:
Even if I make it a static const, MSVC won't optimize it away. I am unimpressed.

Reply:
I see this, but the workaround doesn't work in Compiler Explorer, even with /fp:fast

https://developercommunity.visualstudio.com/t/MSVC-compiler-doesnt-optimize-powx2/10347300?q=%22task+runner%22+gulp+duration&sort=newest

Reply:
Using `constexpr` instead of `const` should guarantee evaluation at compile time, if that's a concern.
I agree with using the full expression instead of evaluating it ourselves and just having a single literal here. It's much more informative than a single value would be, especially with the context of the actual scanning process. And it makes errors in this value easier to spot.
The only drawback I think is that the actual value is not immediately clear when reading it, you'd have to evaluate it yourself, but if it is a `constexpr`, and IDE should be able to tell you what it is anyway.

Reply:
With constexpr, I get an error at compile time since pow and sqrt are not constexpr.

Reply:
The full expression is only executed once at startup, and only takes 14 ns to run on my computer. I see no real reason to have the pre-computed value in the code.

Reply:
From my reading of the assembly language, MSVC only calculates it once at runtime, the first time it's needed. Thus, the performance hit is essentially zero.

https://godbolt.org/z/dG3GKPv1n